Game/Source/GDKTestGyms/GDKTestGymsGameInstance.cpp

```diff
@@ -21,19 +22,50 @@ void UGDKTestGymsGameInstance::OnStart()
 	}
 }
 
-bool UGDKTestGymsGameInstance::Tick(float DeltaSeconds)
+float UGDKTestGymsGameInstance::AddAndCalcFps(int64 NowReal, float DeltaS)
 {
-	float Alpha = 0.8f;
-	float CurrFPS = 1.0f / DeltaSeconds; // TODO - we should remove FApp::GetIdleTime() from DeltaSeconds once the NFR regex is fixed
-	AverageFPS = (Alpha * AverageFPS) + ((1.0f - Alpha) * CurrFPS);
+	int64 DeltaTicks = FTimespan::FromSeconds(DeltaS).GetTicks();
+	TicksForFPS.Push(TPair<int64, int64>(NowReal, DeltaTicks)); 
+	TickWindowTotal += DeltaTicks;
+
+	int64 MinRange = NowReal - FTimespan::FromMinutes(2.0).GetTicks();
+	
+	// Trim samples
+	for(int i = 0; i < TicksForFPS.Num(); i++)
+	{
+		const FPSTimePoint& TimePoint = TicksForFPS[i];
+		if (TimePoint.Key > MinRange) // Find the first valid sample
+		{
+			int NumToRemove = i;
+			if (NumToRemove > 0)
+			{
+				for (int j = 0; j < NumToRemove; j++)
+				{
+					TickWindowTotal -= TicksForFPS[j].Value;
```

Comment:
perhaps we could do: 

if (TimePoint.Key > MinRange) // Find the first valid sample
{
   logic
}
else
{
    TickWindowTotal -= TicksForFPS[i].Value;
}

Reply:
Nit: This is more preference but you might think this makes things easier to read.
I would have NumToRemove declared outside the loop. The loop simply sets NumToRemove and breaks.
After the loop we remove the elements from TicksForFPS. 

Feel free to go with whatever you prefer here. For me my suggestion just breaks things up into two steps and removes some indentation.

Reply:
It was initially like that but it sort of combined 2 steps together which made things slightly unclear. I like the 2 steps, makes sense and breaks it up